service/frontend/workflowHandler.go

```diff
@@ -3225,3 +3267,201 @@ func (wh *WorkflowHandler) getArchivedHistory(
 		Archived:      common.BoolPtr(true),
 	}, nil
 }
+
+func (wh *WorkflowHandler) convertIndexedKeys(keys map[string]interface{}) map[string]gen.IndexedValueType {
+	converted := make(map[string]gen.IndexedValueType)
+	for k, v := range keys {
+		switch v {
+		case gen.IndexedValueTypeString:
+			converted[k] = gen.IndexedValueTypeString
+		case gen.IndexedValueTypeKeyword:
+			converted[k] = gen.IndexedValueTypeKeyword
+		case gen.IndexedValueTypeInt:
+			converted[k] = gen.IndexedValueTypeInt
+		case gen.IndexedValueTypeDouble:
+			converted[k] = gen.IndexedValueTypeDouble
+		case gen.IndexedValueTypeBool:
+			converted[k] = gen.IndexedValueTypeBool
+		case gen.IndexedValueTypeDatetime:
+			converted[k] = gen.IndexedValueTypeDatetime
+		default:
+			wh.GetLogger().Fatal("unknown index value type", tag.Value(v))
```

Comment:
Can we log an error instead of Fatal here?  We want to fail the call with InternalServiceError rather than crashing the host.

Reply:
Updated